dev/checkstyle/suppressions.xml

```diff
@@ -13,7 +13,7 @@
 
     <suppress checks="ParameterNumber" files="CtagsReader\.java|Definitions\.java|PerlLexHelper\.java|
         |JFlexXrefUtils\.java|RubyLexHelper\.java|FileAnalyzerFactory\.java|SearchController\.java|
-        |Context\.java|HistoryContext\.java|Indexer\.java" />
+        |Context\.java|HistoryContext\.java|Indexer\.java|Suggester\.java" />
```

Comment:
I would avoid doing this. In no time we will have telescoping constructor anti-pattern.

Reply:
filed #2610 to track them all. I'd like to get this in without introducing additional semi-complex changes.